Assignment 3

anaaraujo92 opened the pull request.

anaaraujo92 committed:
modified:   life_expectancy/tests/conftest.py
new file:   life_expectancy/tests/new_fixtures.py
modified:   life_expectancy/tests/test_cleaning.py

anaaraujo92 committed:
modified:   life_expectancy/cleaning.py
	new file:   life_expectancy/loadings_savings.py
	modified:   life_expectancy/tests/conftest.py
	new file:   life_expectancy/tests/fixtures/eu_life_expectancy_raw.tsv
	new file:   life_expectancy/tests/fixtures/eu_life_expectancy_raw_sample.tsv
	new file:   life_expectancy/tests/fixtures/eu_life_expectancy_raw_sample_expected.csv
	modified:   life_expectancy/tests/new_fixtures.py
	modified:   life_expectancy/tests/test_cleaning.py
	new file:   life_expectancy/tests/test_load.py
	new file:   life_expectancy/tests/test_save.py

anaaraujo92 committed:
modified:   life_expectancy/tests/new_fixtures.py

anaaraujo92 committed:
modified:   life_expectancy/cleaning.py
new file:   life_expectancy/regions.py
modified:   life_expectancy/tests/conftest.py
modified:   life_expectancy/tests/test_cleaning.py
new file:   life_expectancy/tests/test_regions.py

danizao commented on life_expectancy/loadings_savings.py:
Ana, talvez incluir um descritivo do que faz este arquivo no inicio, para que quem o vê pela primeira vez possa perceber o que o módulo faz sem ter que olhar para as funções?

anaaraujo92 replied:
Certo! adicionei

danizao commented on life_expectancy/loadings_savings.py:
```diff
@@ -0,0 +1,26 @@
+import pandas as pd
+from life_expectancy.cleaning import clean_data
+
+def load_data(path) -> pd.DataFrame:
```
Não queres explicitar que tipo de variável é path? str, certo?

danizao commented on life_expectancy/loadings_savings.py:
Acho também que devias caracterizar as variáveis pelo seu tipo:
- df: Dataframe,
- path: str
- etc

danizao commented on life_expectancy/tests/new_fixtures.py:
Olha, aqui não seria melhor colocares as variáveis como relativas em vez de absolutas?

anaaraujo92 replied:
alterei :)

danizao commented on life_expectancy/tests/new_fixtures.py:
```diff
@@ -0,0 +1,19 @@
+import pandas as pd
+from life_expectancy.cleaning import clean_data
+
+DATA_DIR = Path("C:/Users/afaraujo/OneDrive - Hovione/Documents/DareData/Foundations-LP/assignments/life_expectancy/data")
```
Variáveis relativas em vez de absolutas?

anaaraujo92 replied:
alterado :)

danizao commented on life_expectancy/tests/test_cleaning.py:
```diff
@@ -1,24 +1,20 @@
-TEST_INPUT_PATH = "life_expectancy/data/eu_life_expectancy_raw.tsv"
-
-def test_clean_data():
+def test_clean_data(eu_life_expectancy_raw_sample, eu_life_expectancy_raw_sample_expected):
```
Type of the variables

danizao reviewed (commented):
Well done Ana. Só alguns comentários. Desculpa o atraso na resposta!

anaaraujo92 committed:
modified:   life_expectancy/loadings_savings.py
modified:   life_expectancy/tests/new_fixtures.py
modified:   life_expectancy/tests/test_cleaning.py

anaaraujo92 commented:
> Well done Ana. Só alguns comentários. Desculpa o atraso na resposta!

@danizao , obrigada pelo feeback, fiz as alterações :)

anaaraujo92 committed:
modified:   life_expectancy/tests/new_fixtures.py

anaaraujo92 committed:
modified:   life_expectancy/cleaning.py
